seismicpro/src/seismic_batch.py

```diff
@@ -240,6 +240,82 @@ def trace_headers(self, header, flatten=False):
 
         return np.array(np.split(values, np.cumsum(tracecounts)[:-1]) + [None])[:-1]
 
+    @action
+    def copy_meta(self, from_comp, to_comp, keys=None):
+        """Copy meta from one component to another or form few components. One can copy either
+        full meta or only particular keys.
+
+        Parameters
+        ----------
+        from_comp : str or array-like
+            Component's name to copy meta from or list of component's names.
+        to_comp : str or array-like
+            Component's name to copy meta in or list of component's names.
+        keys : str, array-like or None, optional
+            if None, all meta will be copied
+            if str, only meta[keys] will be copied
+            if array-like,
+                if 1d array-like, for each `to_comp` will be used same keys
+                if 2d array-like, for each `to_comp` will be used particular keys.
+                It means that i-th component will contain only `keys[i]` elements from meta.
+
+        Raises
+        ------
+        ValueError : if `from_comp` and `to_comp` have different length.
+        ValueError : if one of given to `from_comp` component doesn't exist.
+
+        Note
+        ----
+            The copied meta will not overwrite the old meta.
+
+        Examples
+        --------
+        Use all meta:
+            >>> batch.copy_meta('raw', 'new_raw')
+            >>> batch.meta
+            {'raw': {'arg_1': 'val_1', 'arg_2': 'val_2', 'arg_3': 'val_3'},
+            'new_raw': {'arg_1': 'val_1', 'arg_2': 'val_2', 'arg_3': 'val_3'}}
+
+
+        Copy two values:
+            >>> batch.copy_meta('raw', 'new_raw', key=['arg_1', 'arg_2'])
+            >>> batch.meta
+            {'raw': {'arg_1': 'val_1', 'arg_2': 'val_2', 'arg_3': 'val_3'},
+            'new_raw': {'arg_1': 'val_1', 'arg_2': 'val_2'}}
+
+
+        Use for few components:
+            >>> batch.copy_meta(['raw', 'raw'], ['new_raw', 'new_raw_2'], key=[['arg_1'], ['arg_2']])
+            >>> batch.meta
+            {'raw': {'arg_1': 'val_1', 'arg_2': 'val_2', 'arg_3': 'val_3'},
+            'new_raw': {'arg_1': 'val_1'},
+            'new_raw': {'arg_2': 'val_2'}}
+        """
+        from_comp = (from_comp, ) if isinstance(from_comp, str) else from_comp
+        to_comp = (to_comp, ) if isinstance(to_comp, str) else to_comp
+
+        if len(from_comp) != len(to_comp):
+            raise ValueError('Length from_comp should be equal to to_comp length.')
+
+        if isinstance(keys, str) or keys is None:
+            keys = (keys, ) * len(from_comp)
+
+        for ix, (fr_comp, t_comp) in enumerate(zip(from_comp, to_comp)):
+            if fr_comp not in self.meta:
+                raise ValueError(f'{fr_comp} not exist.')
+
+            added_meta = self.meta[fr_comp]
```

Comment:
```suggestion
            added_meta = self.meta[fr_comp].copy()
```
Looks like you can avoid using intermediate dict `new_meta` at all by updating directly `added_meta` bellow